code/game/machinery/scav_selling_pad.dm

```diff
@@ -0,0 +1,205 @@
+/obj/machinery/scav_selling
+	name = "cargo hold pad"
+	icon = 'icons/obj/telescience.dmi'
+	icon_state = "lpad-idle-o"
+	var/idle_state = "lpad-idle-o"
+	var/warmup_state = "lpad-idle"
+	var/sending_state = "lpad-beam"
+	var/cargo_hold_id
+
+/obj/machinery/scav_selling/multitool_act(mob/living/user, obj/item/multitool/I)
+	if (istype(I))
+		to_chat(user, "<span class='notice'>You register [src] in [I]s buffer.</span>")
+		I.buffer = src
+		return TRUE
+
+/obj/machinery/computer/scav_selling_control
+	name = "cargo hold control terminal"
+	var/status_report = "Idle"
+	var/obj/machinery/scav_selling/pad
+	var/warmup_time = 100
+	var/sending = FALSE
+	var/points = 0
+	var/datum/export_report/total_report
+	var/sending_timer
+	var/cargo_hold_id
+
+/obj/machinery/computer/scav_selling_control/Initialize()
+	..()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/machinery/computer/scav_selling_control/multitool_act(mob/living/user, obj/item/multitool/I)
+	if (istype(I) && istype(I.buffer,/obj/machinery/scav_selling))
+		to_chat(user, "<span class='notice'>You link [src] with [I.buffer] in [I] buffer.</span>")
+		pad = I.buffer
+		updateDialog()
+		return TRUE
+
+/obj/machinery/computer/scav_selling_control/LateInitialize()
+	. = ..()
+	if(cargo_hold_id)
+		for(var/obj/machinery/scav_selling/P in GLOB.machines)
+			if(P.cargo_hold_id == cargo_hold_id)
+				pad = P
+				return
+	else
+		pad = locate() in range(4,src)
+
+/obj/machinery/computer/scav_selling_control/ui_interact(mob/user)
```

Comment:
I would strongly prefer a tgui-next UI.

Reply:
No clue how to set up tgui next.
The pirate selling computer uses the exact same ui code btw